libs/sdk-core/src/breez_services.rs

```diff
@@ -473,14 +493,56 @@ impl BreezServices {
         pair_hash: String,
     ) -> Result<ReverseSwapInfo> {
         let routing_hop = self.lsp_info().await?;
-        self.btc_send_swapper
+        let created_reverse_swap_info = self
+            .btc_send_swapper
             .create_reverse_swap(
                 amount_sat,
                 onchain_recipient_address.clone(),
                 pair_hash,
                 routing_hop.pubkey,
             )
-            .await
+            .await?;
+        info!("Created reverse swap: {created_reverse_swap_info:?}");
+
+        // Wait until one of the following happens:
+        // - trying to pay the HODL invoice explicitly fails from Greenlight
+        // - the regular poll of the Breez API detects the status of this reverse swap advanced to LockTxMempool
+        //   (meaning Boltz detected that we paid the HODL invoice)
+        // - the max allowed duration of a payment is reached
+        tokio::select! {
+            pay_thread_res = tokio::time::timeout(
+                Duration::from_secs(self.node_api.config().payment_timeout_sec as u64),
```

Comment:
It seems better to take the config from the current component rather than adding another method on the NodeAPI trait. We can pass the config downstream to the btc_send_swapper.

Reply:
Done 0e5371710b514f5ca3c96e0ac8e7ba31b498509b